tanner/tests/test_mysql_db_helper.py

```diff
@@ -0,0 +1,219 @@
+                if self.returned_result == 1:
+                    await self.handler.delete_db(self.db_name)
+
+            self.loop.run_until_complete(connect())
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_check_db_exists(self, m):
+        self.expected_result = 1
+
+        async def setup():
+            await self.cursor.execute('CREATE DATABASE test_db')
+            await self.conn.commit()
+
+        async def test():
+            self.returned_result = await self.handler.check_db_exists(self.db_name)
+
+        self.loop.run_until_complete(setup())
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_result, self.returned_result)
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_check_no_db_exists(self, m):
+        self.expected_result = 0
+
+        async def test():
+            self.returned_result = await self.handler.check_db_exists(self.db_name)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_result, self.returned_result)
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_setup_db_from_config(self, m):
+        config = {
+            "name": "test_db",
+            "tables": [
+                {
+                    "schema": "CREATE TABLE TEST (ID INTEGER PRIMARY KEY, USERNAME TEXT)",
+                    "table_name": "TEST",
+                    "data_tokens": "I,L"
+                },
+                {
+                    "schema": "CREATE TABLE CREDS (ID INTEGER PRIMARY KEY, EMAIL VARCHAR(15), PASSWORD VARCHAR(15))",
+                    "table_name": "CREDS",
+                    "data_tokens": "I,E,P"
+                }
+            ]
+        }
+
+        def mock_read_config():
+            return config
+
+        self.expected_result = [(('ID', 'int(11)', 'NO', 'PRI', None, ''), ('USERNAME', 'text', 'YES', '', None, '')),
+                                (('ID', 'int(11)', 'NO', 'PRI', None, ''),
+                                 ('EMAIL', 'varchar(15)', 'YES', '', None, ''),
+                                 ('PASSWORD', 'varchar(15)', 'YES', '', None, ''))]
+
+        self.result = []
+        self.handler.read_config = mock_read_config
+        self.handler.insert_dummy_data = AsyncMock()
+
+        async def test():
+            await self.handler.setup_db_from_config()
+
+            for table in config["tables"]:
+                await self.cursor.execute('USE test_db')
+                await self.cursor.execute('DESCRIBE {table_name}'.format(table_name=table["table_name"]))
+                result = await self.cursor.fetchall()
+                self.result.append(result)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.result, self.expected_result)
+        self.handler.insert_dummy_data.assert_called_with('CREDS', 'I,E,P', mock.ANY)
+
```

Comment:
What about test table?

Reply:
I have checked that but `setupdbfromconfig` call is only returning the last table used.

Reply:
I think you should solve these issues yourself because if you don't then it will only waste time. If I hadn't reviewed it on the same day then this issue which you already know how to fix would have taken one more day. So please try harder before raising these small issues.